src/common/mfu_flist_copy.c

```diff
@@ -1284,6 +1322,113 @@ static int mfu_create_files(int levels, int minlevel, mfu_flist* lists,
     return rc;
 }
 
+/* creates hardlinks,
+ * returns 0 on success and -1 on error */
+static int mfu_create_hardlinks(int levels, int minlevel, mfu_flist* lists,
+        const mfu_param_path* srcpath,
+        const mfu_param_path* destpath, mfu_copy_opts_t* mfu_copy_opts)
+{
+    int rc = 0;
+
+    /* determine whether we should print status messages */
+    int verbose = (mfu_debug_level >= MFU_LOG_VERBOSE);
+
+    /* get current rank */
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    /* indicate to user what phase we're in */
+    if (rank == 0) {
+        MFU_LOG(MFU_LOG_INFO, "Linking files.");
+    }
+
+    /* start timer for entie operation */
+    MPI_Barrier(MPI_COMM_WORLD);
+    double total_start = MPI_Wtime();
+    uint64_t total_count = 0;
+
+    int level;
+    for (level = 0; level < levels; level++) {
+        /* time how long this takes */
+        double start = MPI_Wtime();
+
+        /* get list of items for this level */
+        mfu_flist list = lists[level];
+
+        /* iterate over items and set write bit on directories if needed */
```

Comment:
let's update this comment to match the code

Reply:
Done